text/0000-java-17.md

```diff
@@ -0,0 +1,83 @@
+
+- Start Date: 2023-04-19 
+- RFC PR:
+- FOLIO Issue:
+
+# Java 17
+
+## Summary
+
+From Quesnelia (R2 2023) release on version 17 must be used for JDK and JRE.
+
+## Motivation
+
+### FOLIO Java libraries
+
+FOLIO maintains many Java libraries:
+https://wiki.folio.org/display/REL/pom.xml+dependencies
+
+Some of them are Spring based and have already migrated to Java 17
+because Spring no longer supports Java 11.
+
+As FOLIO still officially supports Java 11 all other FOLIO Java
+libraries must be for Java 11. Java 17 modules can use Java 11
+libraries, but Java 11 modules cannot use Java 17 libraries.
+
+Dropping Java 11 support allows Java libraries to upgrade to Java 17.
```

Comment:
> Java 17 modules can use Java 11 libraries, but Java 11 modules cannot use Java 17 libraries.

It is not necessary to wait to drop support for Java 11 to upgrade libraries to Java 17, only set the expectation that to upgrade to the next version of the library the module also needs to upgrade. The previous versions of the library are still available for Java 11 modules (and can be patched if necessary, as the move to Java 17 would be a major revision to the library).

Reply:
> In contrast, a library is used by multiple modules. If the library migrates to Java 17 it can no longer been used by Java 11 modules.

As I understand it, the previous version of the library (that supports Java 11) can still be used (we don't remove them). And that is a decision that can be made on a per module basis (because of the reasons you cited above).

Thus teams can decide when to upgrade to the version of the library that supports Java 17 separate to when support for 11 is removed.

> The RMB comparison is not applicable here. For a given flower release only a single RMB minor version is officially supported.

I accept that presents a challenge in that supporting only the version of RMB that uses Java 17 effectively forces modules to upgrade to Java 17 too.